src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5184,7 +5184,8 @@ private ScanTypeArgumentListKind ScanTypeArgumentList(NameOptions options)
                 case SyntaxKind.AmpersandAmpersandToken: // e.g. `e is A<B> && e`
                 case SyntaxKind.BarBarToken:             // e.g. `e is A<B> || e`
                 case SyntaxKind.AmpersandToken:          // e.g. `e is A<B> & e`
-                case SyntaxKind.OpenBracketToken:        // e.g. `e is A<B>[]`
+                case SyntaxKind.OpenBracketToken         // e.g. `e is A<B>[]`
+                    when this.PeekToken(1).Kind == SyntaxKind.CloseBracketToken || this.PeekToken(1).Kind == SyntaxKind.CommaToken:
```

Comment:
Also checking comma for something like `case Tuple<int, int>[,]`

Reply:
probably it's best to add a test for multi-dimensional arrays as well, but I'm having trouble dumping syntax kinds. `#define PARSING_TEST_DUMP`  doesn't work on it's own, what I'm missing here?